### PR TITLE
Added ISBN13 <-> ISBN10 conversion for indexing in Elastic

### PR DESCRIPTION
**Pull request by lrosenstrom**
```
"identifiedBy": [
                        {
                            "@type": "ISBN",
                            "value": "0070231826",
                        }
```
is on reindex transformed to

```
"identifiedBy": [
                        {
                            "@type": "ISBN",
                            "value": "0070231826",
                            "_str": "ISBN 0070231826"
                        },
                        {
                            "@type": "ISBN",
                            "value": "9780070231825",
                            "_str": "ISBN 9780070231825"
                        }
```
so the document is searchable using both ISBN10 and ISBN13 in ISBN search as well as fritext. 

- The method addTypedThingIdentifier on the Document class was previously unused. Any risks using it?
- One problem now is that both ISBN types are presented when viewing the doc from the search result list (elastic), and only the original one is shown in the full document view (jsonld).

### CODE REVIEW

File: whelk-core/src/main/groovy/whelk/component/ElasticSearch.groovy [hunk 1]
```diff
@@ -250,6 +253,19 @@ class ElasticSearch {
         return framed
     }
 
+    private static void setComputedProperties(Document doc) {
+        List<String> isbnValues = doc.getIsbnValues()
+
+        if (isbnValues.size() == 1) {
```
Comment (olovy) [inline whelk-core/src/main/groovy/whelk/component/ElasticSearch.groovy]: Why limit it to documents with a single ISBN?
See e.g. https://libris.kb.se/r82z67c35265vx0/data.jsonld

Comment (olovy) [reply whelk-core/src/main/groovy/whelk/component/ElasticSearch.groovy]: What about ISBNs in `getIsbnHiddenValues()` (`["@graph", 1, "indirectlyIdentifiedBy"]`) ? Do we want to handle them too?

Comment (lrosenstrom) [reply whelk-core/src/main/groovy/whelk/component/ElasticSearch.groovy]: Now handling multple ISBN values and generating ISBN10/13 for indirectlyIdentifiedBy.

File: whelk-core/src/main/groovy/whelk/component/ElasticSearch.groovy [hunk 2]
```diff
@@ -273,29 +291,20 @@ class ElasticSearch {
     }
 
     private static List<Isbn> getOtherIsbn(List<String> isbnValues) {
```
Comment (niklasl) [inline whelk-core/src/main/groovy/whelk/component/ElasticSearch.groovy]: Perhaps have this just take a String and use like `isbnValues.findResults { getOtherIsbn(it) }`?

Comment (lrosenstrom) [reply whelk-core/src/main/groovy/whelk/component/ElasticSearch.groovy]: Yes, that's a bit cleaner. :gem:

**Pull request merged**